client/maps/action-creators.ts

```diff
@@ -0,0 +1,221 @@
+import {
+  GetMapDetailsPayload,
+  GetMapsPayload,
+  MapInfoJson,
+  MapPreferences,
+  MapSortType,
+  MapVisibility,
+  Tileset,
+  UpdateMapPayload,
+  UpdateMapServerBody,
+} from '../../common/maps'
+import { ThunkAction } from '../dispatch-registry'
+import fetch from '../network/fetch'
+import { apiUrl } from '../network/urls'
+import { openSnackbar } from '../snackbars/action-creators'
+import { ClearMaps } from './actions'
+
+const upload = IS_ELECTRON ? require('./upload').default : null
+
+export function uploadLocalMap(path: string, onMapSelect: (map: MapInfoJson) => void): ThunkAction {
+  return async dispatch => {
+    if (!upload) {
+      return
+    }
+
+    dispatch({
+      type: '@maps/uploadLocalMapBegin',
+      payload: { path },
+    })
+
+    dispatch({
+      type: '@maps/uploadLocalMap',
+      payload: upload(path, apiUrl`maps`).then(({ map }: { map: MapInfoJson }) => {
+        if (onMapSelect) {
+          onMapSelect(map)
+        }
+
+        return map
+      }),
+      meta: { path },
+    })
+  }
+}
+
+export function getMapsList(
```

Comment:
this should almost certainly just take an object directly, especially since you immediately convert it into an object